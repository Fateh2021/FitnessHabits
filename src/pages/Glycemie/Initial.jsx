import firebaseDb from "../../firebaseConfig"
import React, { useState, useEffect } from 'react'
import { add, arrowBack, trash} from "ionicons/icons";
import Modifier from './Modifier'
import {toast} from '../../Toast'
import '../Tab1.css'
import {
  IonContent,
  IonHeader,
  IonPage,
  IonTitle,
  IonToolbar,
  IonCard,
  IonCardContent,
  IonIcon,
  IonFab,
  IonFabButton,
  IonModal
} from "@ionic/react";


const Initial = () => {

  var [glycemie, setGlycemie] = useState({})

  // Reading from db
  useEffect(() => {
    firebaseDb.child('NotificationGlycemie').on('value', snapshot => {
        if (snapshot.val() != null)
            setGlycemie({
            ...snapshot.val()
        })
        snapshot.forEach(function (childSnapshot) {
    });

    })
  }, [])

  const [showModal, setShowModal] = useState(false);
  const [currentId, setCurrentId] = useState('');

  const editAction = (currentId) => {
    setCurrentId(currentId);
    setShowModal(true);
  }

  function supprimer(e){
    toast("Horaire supprimer");
    firebaseDb.database.ref('NotificationGlycemie/' + e).remove()
  }

  return (
        <IonPage>
            <IonHeader className="notGlyChange">
                <IonToolbar>
                    <IonTitle class="titre">Notifications glyc√©mie</IonTitle>
                </IonToolbar>

            <IonFab class="arrow" vertical="top" horizontal="end" slot="fixed">
                <IonFabButton href="/settings" size="small">
                    <IonIcon icon={arrowBack} />
                </IonFabButton>
            </IonFab>
        </IonHeader>
        
        <IonModal isOpen={showModal} cssClass='my-custom-class'>
            <IonHeader>
                <IonToolbar>
                    <IonTitle class="titre">Modifier horaire</IonTitle>
                </IonToolbar>

            <IonFab class="arrow" vertical="top" horizontal="end" slot="fixed">
                <IonFabButton onClick={() => {setShowModal(false)}} size="small">
                    <IonIcon icon={arrowBack} />
                </IonFabButton>
            </IonFab>
        </IonHeader>
            <IonContent>
              <Modifier {...({ currentId, glycemie })} />
            </IonContent>
        </IonModal>

        <IonContent>

          {
            Object.keys(glycemie).map(e => {
              return <IonCard key={e} onClick={() => editAction(e)}>
                  <IonCardContent>
                    <div>
                        <div class="heure" name="heure">{new Date(glycemie[e].heure).toLocaleTimeString([], {hour:'2-digit', minute:'2-digit'})}</div>

                        <div name="notifierAvant">Notifier avant {new Date(glycemie[e].notifierAvant).toLocaleTimeString([], {minute:'2-digit'})} min</div>

                        <div class="jour" name="jour">{glycemie[e].jour.join("  ")}</div>
                    </div>
                        <IonFab class="arrow" vertical="top" horizontal="end" slot="fixed">
                            <IonFabButton class="removeButton" size="small" onClick={() => supprimer(e)} href="/glycemie">
                            <IonIcon icon={trash}/>                            
                            </IonFabButton>
                        </IonFab>

                  </IonCardContent>

              </IonCard>

            })
          }
        </IonContent>

        <IonFab vertical="bottom" horizontal="center" slot="fixed" >
            <IonFabButton href="/glycemieAjout"><IonIcon icon={add}/></IonFabButton>
        </IonFab>

      </IonPage>
  );
};
export default Initial;
