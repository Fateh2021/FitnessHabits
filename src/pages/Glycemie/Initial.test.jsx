 import React from 'react';
 import renderer from 'react-test-renderer';
 import Initial from './Initial.jsx'
 
 
 test('Initial', ()=> {
 const component = renderer.create(<Initial/>);
 const tree = component.toJSON();
 expect(tree).toMatchSnapshot();
 
 });
 
 
