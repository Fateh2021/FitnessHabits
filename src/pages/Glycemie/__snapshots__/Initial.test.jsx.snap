// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial 1`] = `
<div
  className="ion-page"
>
  <ion-header>
    <ion-toolbar>
      <ion-title>
        Notifications glyc√©mie
      </ion-title>
    </ion-toolbar>
    <ion-fab>
      <ion-fab-button
        onClick={[Function]}
      >
        <ion-icon />
      </ion-fab-button>
    </ion-fab>
  </ion-header>
  <ion-content />
  <ion-fab>
    <ion-fab-button
      onClick={[Function]}
    >
      <ion-icon />
    </ion-fab-button>
  </ion-fab>
</div>
`;
